src/btree/bt_delete.c

```diff
@@ -363,7 +363,7 @@ __wt_delete_page_instantiate(WT_SESSION_IMPL *session, WT_REF *ref)
 	 * able to do so.)
 	 */
 	WT_RET(__wt_page_modify_init(session, page));
-	if (btree->modified)
+	if (session->dhandle->checkpoint == NULL)
```

Comment:
@bvpvamsikrishna, in https://github.com/wiredtiger/wiredtiger/blob/develop/src/btree/bt_handle.c#L89-L91 there is this code:

```
 89         /* Checkpoint files are readonly. */
 90         readonly = dhandle->checkpoint != NULL ||
 91             F_ISSET(S2C(session), WT_CONN_READONLY);
```

That is, there is another condition that leads to read-only trees.  That is more unusual, but MongoDB does exercise it (e.g., for queryable backups).  I think this situation of a reference to a deleted child page could occur there as well.

Given that, I think we should add a `WT_BTREE_READONLY` flag to btree handles and set it in `bt_handle.c` based on this condition, then turn the change in `bt_delete.c` into `if (!F_ISSET(btree, WT_BTREE_READONLY)) ...`